src/main/java/core/basesyntax/service/impl/FileWriterImpl.java

```
@@ -0,0 +1,26 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.Constants;
+import core.basesyntax.service.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+
+public class FileWriterImpl implements FileWriter {
+
+    @Override
+    public void write(String resultingReport, String fileName) {
+        String filePath = Constants.RESOURCE_PATH + fileName;
+        Path path = Paths.get(filePath);
+        if (Files.notExists(path)) {
+            try {
+                Files.writeString(path, resultingReport, StandardOpenOption.CREATE);
+                System.out.println("Created new file and wrote header.");
+            } catch (IOException e) {
+                System.err.println("Error writing header: " + e.getMessage());
+            }

```

This `catch` block violates checklist item #6: "Don't ignore exceptions in try-catch blocks". Instead of printing the error message to the console, you should re-throw a `RuntimeException` with an informative message and the original exception. For example: `throw new RuntimeException("Error writing to file: " + filePath, e);`